### PR TITLE
hw-2

### PR DESCRIPTION
**Pull request by AndreiYazadzhi**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>
```
Comment (mfedenko09) [inline pom.xml:73]:
```suggestion
</project>

```

**Review by mfedenko09 — CHANGES_REQUESTED** [review]

**Review by mfedenko09 — APPROVED** [review]

File: log/thread.log [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 87
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 89
+ 10:07:26,242 INFO  MyThread:16 - Thread-0: 88
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 90
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 92
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 93
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 94
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 95
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 96
+ 10:07:26,242 INFO  MyThread:16 - Thread-1: 97
```
Comment (RomanDubovskyi) [inline log/thread.log:99]: this file shouldn't be in PR

File: pom.xml (hunk 1 above)
Comment (RomanDubovskyi) [inline pom.xml:73]: empty line at the end of the file

File: src/main/java/Counter.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+public class Counter {
```
Comment (RomanDubovskyi) [inline src/main/java/Counter.java:1]: same here

File: src/main/java/MyCallable.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    @Override
+    public Integer call() throws Exception {
+        return numbers.stream().mapToInt(Integer::valueOf).sum();
+    }
+
+    public static int getSum(List<Integer> list, int threads) {
+        if (threads <= 0 || threads > LIMIT) {
+            throw new RuntimeException("Incorrect number of threads");
+        }
+        ExecutorService executor = Executors.newFixedThreadPool(threads);
```
Comment (RomanDubovskyi) [inline src/main/java/MyCallable.java:26]: separate this logic into 2 different classes. First is callable impl that performs some atomic task. Second class will manage ExecutorService and use Callable impls inside

File: src/main/java/MyRecursive.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.RecursiveTask;
+
+public class MyRecursive extends RecursiveTask<Integer> {
```
Comment (RomanDubovskyi) [inline src/main/java/MyRecursive.java:7]: name classes according to their purpose

File: src/test/java/CalculationTest.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import java.util.List;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class CalculationTest {
```
Comment (RomanDubovskyi) [inline src/test/java/CalculationTest.java:9]: you should have separate class per each class that you're testing

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**